api/v3/utils.php

```diff
@@ -1333,9 +1333,25 @@ function _civicrm_api3_check_required_fields($params, $daoName, $return = FALSE)
  * @return array
  */
 function _civicrm_api3_basic_get($bao_name, $params, $returnAsSuccess = TRUE, $entity = "", $sql = NULL, $uniqueFields = FALSE) {
-  $query = new \Civi\API\SelectQuery($bao_name, $params, $uniqueFields);
+  $entity = CRM_Core_DAO_AllCoreTables::getBriefName(str_replace('_BAO_', '_DAO_', $bao_name));
+  $options = _civicrm_api3_get_options_from_params($params);
+
+  $query = new \Civi\API\Api3SelectQuery($entity);
+  $query->where = $params;
+  if ($options['is_count']) {
+    $query->select = array('count');
+  }
+  else {
+    $query->select = array_keys(array_filter($options['return']));
+    $query->orderBy = $options['sort'];
+    $query->isFillUniqueFields = $uniqueFields;
+  }
+  $query->limit = $options['limit'];
+  $query->offset = $options['offset'];
+  $query->checkPermissions = CRM_Utils_Array::value('check_permissions', $params, FALSE);
```

Comment:
Re: the failed test. I don't believe it's a problem with the hook infrastructure. Look at this:
  1. Here in `basic_get()`, it calls the constructor (`new Api3SelectQuery()`) and then sets the checkPermissions property.
  2. *Inside* the constructor, it calls `$this->query->where($this->getAclClause(self::MAIN_TABLE_ALIAS, $baoName));`. At that point, `checkPermissions` still has its default value (false).
  3. Inside `getAclClause()`, it short-circuits because `checkPermissions` is false.

I took a stab at this with f62e540b82. It might need a similar patch for Api4SelectQuery.